js/battle.js

```diff
@@ -4761,6 +4761,7 @@ var Battle = (function () {
 					actions += "" + poke.getName() + " fell straight down!";
 					poke.removeVolatile('magnetrise');
 					poke.removeVolatile('telekinesis');
+					if (poke.lastmove === 'fly' || poke.lastmove === 'bounce') poke.sprite.animReset();
```

Comment:
If a Flying-type or levitating mon gets hit with Smack Down, that message is sent. If said mon is currently using Dive or Dig, and still gets hit by Smack Down somehow, the message is still sent (correctly, as the "remove flying type/levitate" effect is still applied).